code/modules/mob/living/carbon/xenomorph/powers.dm

```diff
@@ -341,7 +341,7 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
 		return
 
 	to_chat(src, "<span class='notice'>You start focusing your plasma towards [target].</span>")
-	if(!do_after(src, transfer_delay, TRUE, 5, BUSY_ICON_FRIENDLY))
+	if(!do_after(src, transfer_delay, TRUE, icon_display = USER_ICON_FRIENDLY) || QDELETED(target))
```

Comment:
`do_mob` perhaps?

Reply:
I didn't include the target in the do_after as the ability is still supposed to be usable on moving targets, as long as they don't stray too far. (I may try to make it a callback proc later, though.)